assets/sass/pages/_profile-page.scss

```diff
@@ -0,0 +1,17 @@
+/* profile */
```

Comment:
Putting these rules to a different scss file, but including them to the same "main.css" doesn't help to scope these.

All of these rules could be wrapped to a class for this page, thus they wouln't effect any other page by accident.

Alternatively,  per page styles could be compiled to a separate css file (i.e., just create `page_profile.scss` next to the `main.scss` and the scripts should generate the second css. Of course, that means that you have to import all variables and mixings to the second scss file too.

Any case, for scoping, something like this.

```scss
body.page-profile {
  dt.provider-information {
	text-align: left;
  }
  li.list-group-item.clearfix {
    padding-top: 0px;
    padding-bottom: 0px;
    padding-right: 0px;
  }
}
```

Reply:
Good points!

Reply:
> Putting these rules to a different scss file, but including them to the same "main.css" doesn't help to scope these.

My intention is not to put these rules in a different file to scope them. What I am doing is organizing the [architecture](https://sass-guidelin.es/#architecture) of the scss files.

>All of these rules could be wrapped to a class for this page, thus they wouln't effect any other page by accident.

Yes, this is a much better approach. I will create some classes

> Alternatively, per page styles could be compiled to a separate css file (i.e., just create page_profile.scss next to the main.scss and the scripts should generate the second css. Of course, that means that you have to import all variables and mixings to the second scss file too.

Having one single **main.css** file may help to improve the user experience, as all the styles are loaded only once, and then cached styles are used instead. I thought that was the reason why the CSS was minified.

Reply:
I decided to nest the selectors under the `.local-storage-fields` class. This class is only used in the **profile.html** page

Reply:
> Having one single main.css file may help to improve the user experience, as all the styles are loaded only once, and then cached styles are used instead. I thought that was the reason why the CSS was minified.

In my mind, this is double-edged sword. If everything is in a single file, then every client needs to download a lot of stuff, which isn't ever used. On the other hand, HTTP/1.0 and HTTP/1.1 are not so good with multiple files. Hence, to my best undestanding, the best would be to group files so that both concerns are covered. E.g., for A+, I probably would have all the css used to render student pages in main.css and all the admin extra in an another file. Profile page is 50/50, rarely used, but also doesn't contain many rules.